config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/util/string/CharacterBlacklistUtil.java

```diff
@@ -73,5 +78,58 @@ public static boolean containsInvalidCharacter(String name) {
         return false;
     }
 
+    /**
+     * Returns URL encoded name (even asterisk is encoded)
+     */
+    public static String encodeName(String name) throws UnsupportedEncodingException {
+        String encodedName = URLEncoder.encode(name, "UTF-8");
```

Comment:
Same here. Change it to RuntimeException or just catch and ignore here itself with a WARNING log. This will make the caller code cleaner and non-repetitive